astronomer/providers/sftp/example_dags/example_sftp.py

```diff
@@ -16,6 +46,168 @@
 }
 
 
+def create_sftp_airflow_connection(task_instance: Any) -> None:
+    """
+    Checks if airflow connection exists, if yes then deletes it.
+    Then, create a new sftp_default connection.
+    """
+    conn = Connection(
+        conn_id="sftp_default",
+        conn_type="sftp",
+        host=task_instance.xcom_pull(key="instance_public_dns_name", task_ids=["get_instance_details"])[0],
+        login="ubuntu",
+        port=SFTP_SSH_PORT,
+        extra=json.dumps(
+            {
+                "key_file": f"/usr/local/airflow/dags/{PEM_FILENAME}.pem",
+                "no_host_key_check": "true",
+                "known_hosts": "none",
+            }
+        ),
+    )  # create a connection object
+
+    session = settings.Session()
+    connection = session.query(Connection).filter_by(conn_id=conn.conn_id).one_or_none()
+    if connection is None:
+        logging.info("Connection %s doesn't exist.", str(conn.conn_id))
+    else:
+        session.delete(connection)
+        session.commit()
+        logging.info("Connection %s deleted.", str(conn.conn_id))
+
+    session.add(conn)
+    session.commit()  # it will insert the connection object programmatically.
+    logging.info("Connection sftp_default is created")
+
+
+def create_instance_with_security_group() -> None:
+    """Create ec2 instance"""
+    import boto3
+
+    ec2 = boto3.resource("ec2", **AWS_S3_CREDS)
+    instance = ec2.create_instances(
+        ImageId=AMI_ID,
+        MinCount=1,
+        MaxCount=1,
+        InstanceType=SFTP_INSTANCE_TYPE,
+        KeyName=PEM_FILENAME,
+        SecurityGroups=[INBOUND_SECURITY_GROUP],
+    )
+    instance_id = instance[0].id
+    ti = get_current_context()["ti"]
+    ti.xcom_push(key="ec2_instance_id", value=instance_id)
+
+
+def get_ec2_instance_details(task_instance: "TaskInstance") -> None:
+    """Get the EC2 instance details by id retrieved from the Xcom"""
+    import boto3
+
+    client = boto3.client("ec2", **AWS_S3_CREDS)
+    ec2_instance_id_xcom = task_instance.xcom_pull(key="ec2_instance_id", task_ids=["create_ec2_instance"])[0]
+    response = client.describe_instances(
+        InstanceIds=[ec2_instance_id_xcom],
+    )
+    instance_details = response["Reservations"][0]["Instances"][0]
+    ti = get_current_context()["ti"]
+    ti.xcom_push(
+        key="instance_response_master_security_group", value=instance_details["SecurityGroups"][0]["GroupId"]
```

Comment:
Similarly, let's create a constant for `instance_response_master_security_group`